openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
+            } else if ( operator == "Weighted multiplication") {
+                for (var p4 in la[o].properties) {
+                    // iterate over the indicator child keys
+                    for (var r4 = 0; r4 < indicatorChildrenKey.length; r4++, ct++) {
+                        if (p4 == indicatorChildrenKey[r4]) {
+                            // Sum the theme indicators
+                            var weight = tempChildren[r4].weight;
+                            if (tempSum == 0) {
+                                tempSum = (la[o].properties[p4] * weight);
+                            } else {
+                                tempSum = tempSum * (la[o].properties[p4] * weight);
+                            }
+                        }
                     }
                 }
-                scaleCIvalues = [];
-                tempCIvalues = [];
-            }
-
-            // The data passed into d3 need to be an array of objects
-            // Place the category indicator objects into an array
-            var catData = [];
-
-            for (var k in scaledCatIndicator) {
-                catData.push(scaledCatIndicator[k]);
-            }
-
-            // TODO remove this concat if need be
-            var concat = catData.concat(previousCatData);
-
-            Category_PCP_Chart(catData, municipality, districName, concat);
-
-            for (var k in scaledCatIndicator) {
-                previousCatData.push(scaledCatIndicator[k]);
-            }
-
-            /////////////////////////////////////////////
-            /////////// Create the svi object ///////////
-            /////////////////////////////////////////////
-
-            for (var k in sviIndicator) {
-                delete sviIndicator[k];
-            }
-
-            for (var k in scaledSviIndicator) {
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
+                var munic4 = la[o].properties[selectedRegion];
+                var theme4 = name;
+                tempString.push(munic4 + '|'+ theme4 +'|'+ tempSum);
             }
+        }
+    }
 
-            // Associate the newSVIvalues method with the svi object
-            sviIndicator.SVIvalues = newSVIvalues;
-            var tempSviParentChildKey = [];
-            var tempSviIndicator = {};
-            for (var i = 0; i < municipality.length; i++) {
-                tempSviIndicator[municipality[i]] = [];
-            }
-            tempSviSearchElements = Object.keys(weightedCatIndicator);
-            tempSviSearchElements.pop();
-            var sessionKey = Object.keys(weightedCatIndicator[tempSviSearchElements[0]]);
-            sessionKey.pop();
-            sessionKey.pop();
+    for (var p5 = 0; p5 < tempString.length; p5++) {
```

Comment:
If you build a structured object as suggested, it would be much clearer here to get the information from that, instead of parsing the string.